src/compile/scale/domain.ts

```diff
@@ -314,60 +312,97 @@ export function canUseUnaggregatedDomain(fieldDef: FieldDef<string>, scaleType:
 }
 
 /**
- * Convert the domain to an array of data refs or an array of values. Also, throw
- * away sorting information since we always sort the domain when we union two domains.
+ * Converts an array of domains to a single Vega scale domain.
  */
-function normalizeDomain(domain: VgDomain): (any[] | VgDataRef | VgSignalRef)[] {
-  if (util.isArray(domain)) {
-    return [domain];
-  } else if (isSignalRefDomain(domain)) {
-    return [domain];
-  } else if (isDataRefDomain(domain)) {
-    delete domain.sort;
-    return [domain];
-  } else if(isFieldRefUnionDomain(domain)) {
-    return domain.fields.map(d => {
+export function mergeDomains(domains: VgNonUnionDomain[]): VgDomain {
+  const uniqueDomains = util.unique(domains.map(d => {
+    // ignore sort property when computing the unique domains
+    if (isDataRefDomain(d)) {
+      const {sort: _s, ...dom} = d;
+      return dom;
+    }
+    return d;
+  }), util.hash);
+
+  const sorts: VgSortField[] = util.unique(domains.map(d => {
+    if (isDataRefDomain(d)) {
+      const s = d.sort;
+      if (s !== undefined && !isBoolean(s)) {
+        if (s.op === 'count') {
+          // let's make sure that if op is count, we don't use a field
+          delete s.field;
+        }
+        if (s.order === 'ascending') {
+          // drop order: ascending as it is the default
+          delete s.order;
+        }
+      }
+      return s;
+    }
+    return undefined;
+  }).filter(s => s !== undefined), util.hash);
+
+  if (uniqueDomains.length === 1) {
+    const domain = domains[0];
+    if (isDataRefDomain(domain) && sorts.length > 0) {
+      let sort = sorts[0];
+      if (sorts.length > 1) {
+        log.warn(log.message.MORE_THAN_ONE_SORT);
+        sort = true;
+      }
       return {
-        data: domain.data,
-        field: d
+        ...domain,
+        sort: sort
```

Comment:
huh?    `{sort: sort}` is equivalent to `{sort}`.    So I'm not sure what you're talking about.